src/core/Table/hooks/useResizeColumns.tsx

```diff
@@ -0,0 +1,440 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+import React from 'react';
+import {
+  actions,
+  ActionType,
+  ColumnInstance,
+  HeaderGroup,
+  Hooks,
+  makePropGetter,
+  TableInstance,
+  TableKeyedProps,
+  TableState,
+  useGetLatest,
+} from 'react-table';
+
+export const useResizeColumns = <T extends Record<string, unknown>>(
+  ownerDocument: Document | undefined,
+) => (hooks: Hooks<T>) => {
+  hooks.getResizerProps = [defaultGetResizerProps(ownerDocument)];
+  hooks.stateReducers.push(reducer);
+  hooks.useInstanceBeforeDimensions.push(useInstanceBeforeDimensions);
+};
+
+const defaultGetResizerProps = (ownerDocument: Document | undefined) => (
+  props: TableKeyedProps,
+  {
+    instance,
+    header,
+    nextHeader,
+  }: {
+    instance: TableInstance;
+    header: HeaderGroup;
+    nextHeader: HeaderGroup;
+  },
+) => {
+  if (!ownerDocument) {
+    return props;
+  }
+
+  const { dispatch, flatHeaders } = instance;
+
+  const onResizeStart = (
+    e: React.TouchEvent & React.MouseEvent,
+    header: HeaderGroup,
+  ) => {
+    let isTouchEvent = false;
+    if (e.type === 'touchstart') {
+      // lets not respond to multiple touches (e.g. 2 or 3 fingers)
+      if (e.touches && e.touches.length > 1) {
+        return;
+      }
+      isTouchEvent = true;
+    }
+
+    // Setting `width` here because it might take several rerenders until actual column width is set.
+    flatHeaders.forEach((h) => {
+      if (!h.width) {
+        h.width = h.resizeWidth;
+      }
+    });
+
+    const headerIdWidths = getLeafHeaders(header).map((d) => [d.id, d.width]);
+    const nextHeaderIdWidths = nextHeader
+      ? getLeafHeaders(nextHeader).map((d) => [d.id, d.width])
+      : [];
+
+    const clientX = isTouchEvent ? Math.round(e.touches[0].clientX) : e.clientX;
+
+    const dispatchMove = (clientXPos: number) =>
+      dispatch({ type: actions.columnResizing, clientX: clientXPos });
```

Comment:
is this cought by react-table and need only clientx? or this might go to the user too..? in that case, could there be a situation, that we need more than just clientX .. ?

Reply:
We talked about this in chat that `dispatch` is being used to handle internal `Table` state.